client/src/context/useKanbanContext.tsx

```diff
@@ -1,45 +1,84 @@
+import { useAuth } from './useAuthContext';
 
 export const KanbanContext = createContext<IKanbanContext>({} as IKanbanContext);
 
 export const KanbanProvider: FunctionComponent = ({ children }): JSX.Element => {
-  const [columns, setColumns] = useState<Array<IColumn>>(columnData);
+  const [activeBoard, setActiveBoard] = useState<IBoard>({
+    _id: 'Initial',
+    name: 'Initial',
+    columns: [],
+    cards: [],
+    user: 'Initial',
+    createdAt: 'Initial',
+  });
+  const [columns, setColumns] = useState<IColumn[]>(activeBoard.columns);
   const [focusedCard, setFocusedCard] = useState<ICard | null>(null);
   const { updateSnackBarMessage } = useSnackBar();
+  const { loggedInUser } = useAuth();
+
+  useEffect(() => {
+    if (loggedInUser) getFirstBoard();
+
+    return;
+  }, [loggedInUser]);
+
+  // Helper function to set the first board;
+  const getFirstBoard = async () => {
+    const request = await getUserBoards();
+    const board = request.boards[0];
+    setActiveBoard(board);
+    setColumns(board.columns);
+
+    return board;
+  };
+
   const handleDragEnd = (result: DropResult): void => {
+    if (!result.destination || !columns) return;
+
     const { destination, source, draggableId } = result;
-    if (!destination) return;
+
     const columnsCopy: IColumn[] = cloneDeep(columns);
-    const colIndex = columns.findIndex((col) => col.id === source.droppableId);
-    if (source.droppableId === destination.droppableId) {
-      if (colIndex > -1) {
-        const cards = Array.from(columnsCopy[colIndex].cards);
-        const newCards = swapCards(cards, source, destination, draggableId);
-        columnsCopy[colIndex].cards = newCards;
-        setColumns(columnsCopy);
-        return;
-      }
+    const colIndex = columns.findIndex((col) => col._id === source.droppableId);
+
+    if (source.droppableId === destination.droppableId && colIndex > -1) {
+      const cards = Array.from(columnsCopy[colIndex].cards);
+      const newCards = swapCards(cards, source, destination, draggableId);
+      columnsCopy[colIndex].cards = newCards;
+      activeBoard.columns = columnsCopy;
+
+      setColumns(columnsCopy);
+
+      updateBoard(activeBoard);
+
+      return;
     }
 
     if (source.droppableId !== destination.droppableId) {
-      const targetColumnIndex = columnsCopy.findIndex((col) => col.id === destination.droppableId);
+      const targetColumnIndex = columnsCopy.findIndex((col) => col._id === destination.droppableId);
       if (targetColumnIndex > -1) {
         const targetColumn = columnsCopy[targetColumnIndex];
         const originalColumn = columnsCopy[colIndex];
         const [card] = originalColumn.cards.splice(source.index, 1);
-        card.columnId = targetColumn.id;
+        card.columnId = targetColumn._id;
         targetColumn.cards.splice(destination.index, 0, card);
       }
     }
 
+    activeBoard.columns = columnsCopy;
+
     setColumns(columnsCopy);
+
+    updateBoard(activeBoard);
```

Comment:
The meat of the board's logic; `activeBoard.columns = columnsCopy;` occurs so it could be displayed on the frontend, `updateBoard` occurs to persist data beyond the frontend